### PR TITLE
e2e tests for DevTools package

### PR DESCRIPTION
**Pull request by christian-bromann**
To ensure that the DevTools driver is running according the WebDriver spec we should run some e2e test. Given that having an e2e test for every single command with all edge cases would be a bit to much therefor I only cover the most important commands (ignore things like timeouts, getWindowRect etc).

Yes, this will add another test to the pipeline but since it is running in headless mode it runs pretty fast and only adds 11.361s to the overall test.

### CODE REVIEW

**Comment by christian-bromann** [comment]
@webdriverio/project-committers this is ready for review

File: e2e/devtools.test.js [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+    })
+
+    it('findElements with css selector', async () => {
+        const boxes = await browser.findElements('css selector', '.box')
+        expect(boxes).toHaveLength(5)
+    })
+
+    it('find nested elements with css selector', async () => {
+        const nested = await browser.findElement('css selector', '.nested')
+        const spans = await browser.findElementsFromElement(nested[ELEMENT_KEY], 'css selector', 'span')
```
Comment (mgrybyk) [inline e2e/devtools.test.js]: what do you think on adding `findElementsFromElement` for xpath strategy as well?

File: packages/devtools/src/commands/elementClick.js [hunk 2]
```diff
@@ -29,11 +30,27 @@ export default async function elementClick ({ elementId }) {
...
+         */
+        page.once('framenavigated', () => {
+            clearTimeout(waitForPageLoadTimeout)
+            page.once('load', () => resolve(null))
+        })
+
         const dialogHandler = () => resolve()
         page.once('dialog', dialogHandler)
         return elementHandle.click().then(() => {
             page.removeListener('dialog', dialogHandler)
```
Comment (mgrybyk) [inline packages/devtools/src/commands/elementClick.js]: should we remove `framenavigated` as well?

File: packages/devtools/src/commands/findElement.js [hunk 3]
```diff
@@ -8,9 +8,17 @@ export default async function findElement ({ using, value }) {
         throw new Error(`selector strategy "${using}" is not yet supported`)
     }
 
-    const page = this.windows.get(this.currentWindowHandle)
+    const page = this.getPageHandle()
     let needsCleanUp = false
 
+    if (using === 'link text') {
+        using = 'xpath'
+        value = `.//a[normalize-space() = "${value}"]`
```
Comment (mgrybyk) [inline packages/devtools/src/commands/findElement.js]: why do we need `.` here? There should be no nested element when using findElement(s), I think.

File: e2e/devtools.test.js [hunk 4]
```diff
@@ -0,0 +1,385 @@
+import DevTools from '../packages/devtools/src/index'
```
Comment (mgrybyk) [inline e2e/devtools.test.js]: what do you think on splitting this files into several ones to run tests in parallel?

Comment (christian-bromann) [reply e2e/devtools.test.js]: I wasn't sure how Jest is doing parallelization but it seems to work 👍 split test suite into 2 files

**Comment by christian-bromann** [comment]
@mgrybyk addressed comments

**Comment by christian-bromann** [comment]
any further comments?

File: packages/devtools/src/commands/elementClick.js [hunk 5]
```diff
@@ -35,15 +35,17 @@ export default async function elementClick ({ elementId }) {
         /**
          * check if page load has happened due to click
          */
-        page.once('framenavigated', () => {
+        const frameNavigatedHandler = () => {
             clearTimeout(waitForPageLoadTimeout)
```
Comment (mgrybyk) [inline packages/devtools/src/commands/elementClick.js]: @christian-bromann  should `dialog` event be removed here as well?

Comment (christian-bromann) [reply packages/devtools/src/commands/elementClick.js]: Ahh, I completely forgot why I had implemented it that way. I added comments to better explain how this is suppose to work.

File: packages/devtools/src/commands/elementClick.js [hunk 6]
```diff
@@ -29,11 +30,43 @@ export default async function elementClick ({ elementId }) {
...
         page.once('dialog', dialogHandler)
         return elementHandle.click().then(() => {
+            /**
+             * no modals popped up, so clean up the listener
+             */
             page.removeListener('dialog', dialogHandler)
-            resolve(null)
+
+            /**
+             * wait for at least 150ms to see if a page load was triggered,
```
Comment (mgrybyk) [inline packages/devtools/src/commands/elementClick.js]: does it mean that we'll wait for 150ms after every click?

Comment (christian-bromann) [reply packages/devtools/src/commands/elementClick.js]: Yes .. I know it is not ideal but the only way we can guarantee we have stable page transitions

Comment (mgrybyk) [reply packages/devtools/src/commands/elementClick.js]: I don't know how to handle it better neither, we may improve it later on

**Comment by mgrybyk** [comment]
LGTM but please fix failing smoke tests
